player_openal.go

```diff
@@ -205,11 +204,21 @@ func (p *player) Close() error {
 		p.bufs = append(p.bufs, bs...)
 	}
 
-	C.alcCloseDevice((*C.struct_ALCdevice_struct)(unsafe.Pointer(p.alDevice)))
-	p.isClosed = true
+	C.alSourceStop(p.alSource)
+	C.alDeleteSources(1, &p.alSource)
+	C.alDeleteBuffers(numBufs, &p.bufs[0])
+	C.alcDestroyContext((*C.struct_ALCcontext_struct)(unsafe.Pointer(p.alContext)))
+
 	if err := getError(p.alDevice); err != nil {
 		return fmt.Errorf("oto: CloseDevice: %v", err)
 	}
+
+	b := C.alcCloseDevice(p.alDevice)
+	if b == C.ALC_FALSE {
+		return fmt.Errorf("oto: CloseDevice: %s failed to close", p.alDeviceName)
```

Comment:
I wasn't sure if it would know what to do or not, fmt's broad support has surprised me before. I've changed it to a GoString though.